packages/markdown/src/vaadin-markdown.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * @license
+ * Copyright (c) 2017 - 2025 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import { css, html, LitElement } from 'lit';
+import { defineCustomElement } from '@vaadin/component-base/src/define.js';
+import { ElementMixin } from '@vaadin/component-base/src/element-mixin.js';
+import { PolylitMixin } from '@vaadin/component-base/src/polylit-mixin.js';
+import { ThemableMixin } from '@vaadin/vaadin-themable-mixin/vaadin-themable-mixin.js';
+
+/**
+ * `<vaadin-markdown>` is a web component for rendering Markdown content.
+ * It takes Markdown source as input and renders the corresponding HTML.
+ *
+ * ### Styling
+ *
+ * The component does not have specific shadow DOM parts for styling the rendered Markdown content itself,
+ * as the content is rendered directly into the component's light DOM via a slot.
+ * You can style the rendered HTML elements using standard CSS selectors targeting the `vaadin-markdown` element.
+ *
+ * See [Styling Components](https://vaadin.com/docs/latest/styling/styling-components) documentation.
+ *
+ * @customElement
+ * @extends HTMLElement
+ * @mixes ElementMixin
+ * @mixes ThemableMixin
+ */
+class Markdown extends ElementMixin(ThemableMixin(PolylitMixin(LitElement))) {
+  static get is() {
+    return 'vaadin-markdown';
+  }
+
+  static get styles() {
+    return css`
+      :host {
+        display: block;
+      }
+
+      :host([hidden]) {
+        display: none;
```

Comment:
```suggestion
        display: none !important;
```

Reply:
I wonder if the `!important` is needed. The element seems to get properly hidden with the attribute in place. I added a test case to ensure it. Same styles are also use in [vaadin-lit-combo-box-item](https://github.com/vaadin/web-components/blob/c8a33608cf1ab377529afd0650360e349886384c/packages/combo-box/src/vaadin-lit-combo-box-item.js#L29-L35).

Reply:
Ah, right, `!important` is required to override external `display` styles. Added a test case and fixed.

Reply:
It's probably not the most common case for such component, but if user changes `display` to something like `inline-block` then `hidden` styles without `!important` would not work anymore.